raster-foundry/raster-foundry
# Add more statuses  for ingest control flow

## Overview

Someday we'll have as many statuses as AWS Batch.

This PR adds a `QUEUED` ingest status for ingests that we know we want to kick off that we haven't
kicked off yet. It also, because we want to update ingest statuses after kicking off ingests, moves that
logic into the `db` subproject so that it can all live somewhere reasonable without the common project
needing to know about the db project again.

It _also_ dramatically reduces the number of places where ingests get kicked off and adds logging
every time we kick off ingests in the db project.

### Checklist

- [x] Any new SQL strings have tests (all exercised in existing tests)

## Testing Instructions

 * start up all ur stuff
 * add a scene to a project
 * watch your api-server logs -- they'll log that they're kicking off an ingest for a scene with ingest status QUEUED
 * don't do the ingest
 * add another
 * note that you only get one logging message for the new scene
 * note that now almost all kicking off ingests goes through the `ProjectDao.addScenesToProject` path
 * confirm logic in `SceneDao.update` makes sense
 * confirm logic in `SceneDao.insert` makes sense

Closes #3990
